slave/src/main/java/org/fao/geonet/readonly/InterceptsStartup.java

```diff
@@ -0,0 +1,287 @@
+	private @Value("${slave.masterURL}") String masterURL;
+	private @Value("${slave.masterNode}") String masterNode;
+	private @Value("${slave.frequency}") String frequency;
+
+	private final String key = "MASTER";
+	private final Logger log = LoggerFactory.getLogger(Geonet.GEONETWORK);
+
+	public void onApplicationEvent(ServerStartup arg0) {
+		removeHarvesters(arg0);
+		addHarvester(arg0);
+		removeUsers(arg0);
+	}
+
+	/**
+	 * Remove all harvesters that don't have the name "MASTER"
+	 * 
+	 * @param arg0
+	 */
+	@Transactional(value = TxType.REQUIRES_NEW)
+	private void removeHarvesters(ServerStartup arg0) {
+		if (removeHarvesters) {
+			HarvesterSettingRepository settingsRepo = arg0.getContext().getBean(HarvesterSettingRepository.class);
+
+			List<HarvesterSetting> harvesters = settingsRepo.findAllChildren(1);
+			for (HarvesterSetting h : harvesters) {
+				for (HarvesterSetting hs : settingsRepo.findAllChildren(h.getId())) {
+					if (hs.getName().equals("site")) {
+						for (HarvesterSetting s : settingsRepo.findAllChildren(hs.getId())) {
+							if (s.getName().equals("name") && !s.getValue().equals(key)) {
+								try {
+									HarvestManager hm = arg0.getContext().getBean(HarvestManager.class);
+									hm.remove(String.valueOf(h.getId()));
+								} catch (Exception e) {
+									log.error("Error removing old harvester in readonly instance: " + h, e);
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Remove all users with an ID above 1 (the default admin user we need to run
+	 * GeoNetwork).
+	 * 
+	 * @param arg0
+	 */
+	@Transactional(value = TxType.REQUIRES_NEW)
+	private void removeUsers(ServerStartup arg0) {
+		if (removeUsers) {
+			userRepository.deleteAll(new Specification<User>() {
+				@Override
+				public Predicate toPredicate(Root<User> root, CriteriaQuery<?> query, CriteriaBuilder cb) {
+					Path<Integer> userIdAttributePath = root.get(User_.id);
+					Predicate userIdEqualPredicate = cb.gt(userIdAttributePath, cb.literal(1));
+					return userIdEqualPredicate;
+				}
+			});
+		}
+	}
+
+	/**
+	 * Create the "MASTER" harvester to replicate all content
+	 * 
+	 * @param arg0
+	 */
+	private void addHarvester(ServerStartup arg0) {
+		int id = harvesterAlreadyCreated(arg0);
+		
+		if (id < 0) {
+
+			Element harvester = new Element("node");
+			harvester.setAttribute("id", "");
+			harvester.setAttribute("type", "geonetwork");
+
+			Element ownerGroup = new Element("ownerGroup");
+			addElementWithContentString(ownerGroup, "id", "2");
+			harvester.addContent(ownerGroup);
+
+			Element ownerUser = new Element("ownerUser");
+			addElementWithContentString(ownerUser, "id", "undefined");
+			harvester.addContent(ownerUser);
+
+			harvester.addContent(createSite());
+			harvester.addContent(createEmptySearches());
+			harvester.addContent(createOptions());
+			harvester.addContent(createContent());
+			harvester.addContent(createPrivileges());
```

Comment:
If you use MEF format, it will copy privileges from Master to Slave including on private groups. So maybe you don't need this ?